new-website/src/components/location/location-table.jsx

```diff
@@ -2,51 +2,93 @@ import React, { useState } from "react"
 import firebase from "gatsby-plugin-firebase"
 import MaterialTable from "material-table"
 import { LOCATION_COLLECTION } from "."
-import { useIntl } from "gatsby-plugin-intl"
+import { navigate, useIntl } from "gatsby-plugin-intl"
 import { generateI18N } from "./table-i18n"
 import { Roller } from "react-spinners-css"
+import shortid from "shortid"
 
-function choosePartnerId(entryPartnerId, userPartnerIds = []) {
-    if (entryPartnerId && userPartnerIds.includes(entryPartnerId)) {
-        return entryPartnerId
+function choosePartnerId(entryPartnerId, userPartnerIds, onError) {
+    if (entryPartnerId) {
+        if (userPartnerIds.includes(entryPartnerId)) {
+            return entryPartnerId
+        }
+        // TODO: Issue this error to sentry?
+        onError({ code: "choosePartnerId/inaccessible" })
+        throw new Error("Provided partner ID inaccessible for user")
```

Comment:
Here I mimicked your approach from below. Why would we need both, calling the error handler and throwing?

Reply:
I think the throwing was left over ... can be removed ... if you return here ... otherwise the code will continue to exceute

Reply:
Ok, so the error hander is not a trap, but it reports the error and returns? Then we may want to rename it to `reportError`?

In this case there seems to be no useful return value and returning a dummy, I had to handle it again up-stack again. So I'd stay with the `throw` and drop the `onError`.

Just out of interest, what is your experience: Is there a significant performance penalty for throwing here (in JS)?

Reply:
No thats fine ... but the on error function enables user notification of any errors ...

Reply:
throw error will just crash the page if it is not  handled properly